File: backend/app/schemas/result.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.question import AnswerOption
from app.schemas.question import QuestionWithAnswer


class ResultSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    quiz_id: int
    question_count: int
    correct_count: int
    score_percentage: float
    started_at: datetime
    completed_at: datetime | None
    time_taken_seconds: int | None


class ReviewItem(BaseModel):
    question_order: int
    selected_answer: AnswerOption | None
    is_correct: bool | None
    flagged: bool
    question: QuestionWithAnswer


class ReviewResponse(BaseModel):
    quiz_id: int
    items: list[ReviewItem]

File: frontend/src/types/Result.ts
import type { AnswerOption, QuestionWithAnswer } from './Question'

export interface ResultSummary {
  quiz_id: number
  question_count: number
  correct_count: number
  score_percentage: number
  started_at: string
  completed_at: string | null
  time_taken_seconds: number | null
}

export interface ReviewItem {
  question_order: number
  selected_answer: AnswerOption | null
  is_correct: boolean | null
  flagged: boolean
  question: QuestionWithAnswer
}

export interface ReviewResponse {
  quiz_id: number
  items: ReviewItem[]
}
